pwndbg/commands/msr.py

```diff
@@ -0,0 +1,181 @@
+        except ValueError:
+            if msr not in COMMON_MSRS[arch]:
+                print(f"unknown MSR {msr} on {arch}")
+                return None
+
+            msr_val = COMMON_MSRS[arch][msr]
+
+    return msr_val
+
+
+def parse_range(msr_range: str, arch: str) -> Tuple[int, int]:
+    bounds = msr_range.split("-")
+    if len(bounds) != 2:
+        return None
+
+    start = parse_msr(bounds[0], arch)
+    end = parse_msr(bounds[1], arch)
+
+    if start is None or end is None:
+        return None
+
+    if start > end:
+        return None
+
+    return (start, end)
+
+
+def x86_msr_read(msr: int) -> None:
+    async def ctrl(ec: pwndbg.dbg_mod.ExecutionController):
+        sc = pwnlib.asm.asm(f"mov ecx, {msr}; rdmsr")
+        async with pwndbg.aglib.shellcode.exec_shellcode(ec, sc):
+            edx = int(pwndbg.aglib.regs["edx"]) << 32
+            eax = int(pwndbg.aglib.regs["eax"])
+            ret = edx + eax
+            print(f"{hex(msr)}:\t{hex(ret)}")
+
+    pwndbg.dbg.selected_inferior().dispatch_execution_controller(ctrl)
+
+
+def x86_msr_write(msr: int, write_value: int) -> None:
+    async def ctrl(ec: pwndbg.dbg_mod.ExecutionController):
+        eax = write_value & 0xFFFFFFFF
+        edx = write_value >> 32
+        sc = pwnlib.asm.asm(f"mov ecx, {msr}; mov eax, {eax}; mov edx, {edx}; wrmsr")
+        async with pwndbg.aglib.shellcode.exec_shellcode(ec, sc):
+            return
+
+    pwndbg.dbg.selected_inferior().dispatch_execution_controller(ctrl)
+
+
+def msr_read(msr: int) -> None:
+    arch = pwndbg.aglib.arch.name
+
+    if arch == "x86" or arch == "x86-64":
+        x86_msr_read(msr)
+    else:
+        print(f"{arch} not supported")
+
+
+def msr_write(msr: int, write_value: int) -> None:
+    arch = pwndbg.aglib.arch.name
+
+    if arch == "x86" or arch == "x86-64":
+        x86_msr_write(msr, write_value)
+    else:
+        print(f"{arch} not supported")
+
+
+def msr_list(arch: str):
+    msrs = COMMON_MSRS[arch]
+    longest_msr_name = max(map(len, msrs))
+    for msr, value in msrs.items():
+        print(f"{msr.ljust(longest_msr_name)} = {hex(value)}")
+
+
+parser = argparse.ArgumentParser(
+    description="""
+Read or write to Model Specifc Register (MSR)
+""",
+)
+parser.add_argument("msr", help="MSR value or name", type=str, nargs="?", default=None)
+parser.add_argument(
+    "-w",
+    "--write",
+    metavar="write_value",
+    help="value to write in MSR",
+    type=int,
+    nargs="?",
+    default=None,
+)
+parser.add_argument(
+    "-l",
+    "--list",
+    dest="list_msr",
+    action="store_true",
+    help="list common MSRs for the current arch",
+    default=False,
+)
+parser.add_argument(
+    "-r",
+    "--range",
+    dest="msr_range",
+    help="dash separated range of MSRs to read (eg. --range=1-10 where 10 is included)",
+    type=str,
+    nargs="?",
+    default=None,
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.KERNEL, command_name="msr")
```

Comment:
```suggestion
@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.KERNEL)
```
command_name is default like your function name